bundles/org.openhab.binding.kvv/src/main/java/org/openhab/binding/kvv/internal/KVVStationHandler.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.kvv.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TimerTask;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * KVVStationHandler represents a station and holds information about the trains
+ * which will arrive soon.
+ *
+ * @author Maximilian Hess - Initial contribution
+ */
+@NonNullByDefault
+public class KVVStationHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(KVVStationHandler.class);
+
+    @Nullable
+    private ScheduledFuture<?> pollingJob;
+
+    public KVVStationHandler(final Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        final KVVStationConfig config = getConfigAs(KVVStationConfig.class);
+        if (config == null) {
+            logger.warn("Failed to get config (is null)");
+            updateStatus(ThingStatus.OFFLINE);
+            return;
+        }
+
+        final Bridge bridge = getBridge();
+        if (bridge == null) {
+            logger.warn("Failed to get bridge (is null)");
+            updateStatus(ThingStatus.OFFLINE);
+            return;
+        }
+
+        final KVVBridgeHandler handler = (KVVBridgeHandler) bridge.getHandler();
+        if (handler == null) {
+            logger.warn("Failed to get bridge handler (is null)");
+            updateStatus(ThingStatus.OFFLINE);
+            return;
+        }
+
+        // creating channels
+        final List<Channel> channels = new ArrayList<Channel>();
```

Comment:
Would it maybe make more sense to use a station as a bridge (maybe put the settings in a global configuration but it could then live on the station).

What you could do then is auto-discover the trains as a Thing-s under the station-Bridge. They would just have normal channels. I think this would make the binding less magic and easier to use.